### PR TITLE
Added All notice route

### PR DESCRIPTION
**Pull request by AJAmran**
I have completed all routes for notice

### CODE REVIEW

File: server/src/models/Notice.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { Schema, model, Document } from "mongoose";
+
+interface INotice extends Document {
```
Comment (UpalBarua) [inline server/src/models/Notice.ts]: We are choosing to use Types over interfaces, so please refactor the interface into a type.

File: server/src/models/Notice.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  status: string;
+  relatedLinks: {
+    title: string;
+    url: string;
+  }[];
+}
+
+const noticeSchema = new Schema<INotice>(
+  {
+    title: {
```
Comment (UpalBarua) [inline server/src/models/Notice.ts:20]: Fields such as 'title' and 'description' should incorporate validations such as minLength, maxLength, and additional criteria.

File: server/src/controllers/noticeController.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      pdfLink,
+      postedDate,
+      expiryDate,
+      author,
+      tags,
+      status,
+      relatedLinks,
+    } = req.body;
+
+    if (
```
Comment (UpalBarua) [inline server/src/controllers/noticeController.ts]: Don't manually validate the body data; use Zod instead.

File: server/src/controllers/noticeController.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      !tags ||
+      !status ||
+      !relatedLinks
+    ) {
+      return res
+        .status(400)
+        .json({ message: "Please provide all required notice fields." });
+    }
+
+    const notice = new Notice({
```
Comment (UpalBarua) [inline server/src/controllers/noticeController.ts]: Use create() method instead

File: server/src/controllers/noticeController.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      pdfLink,
+      postedDate,
+      expiryDate,
+      author,
+      tags,
+      status,
+      relatedLinks,
+    } = req.body;
+
+    if (
```
Comment (UpalBarua) [inline server/src/controllers/noticeController.ts]: Don't validate the body data manually, use Zod instead

**Review by UpalBarua — CHANGES_REQUESTED** [review]
Great code! However, there are some changes needed.

File: server/src/controllers/noticeController.ts (hunk 3 above)
Comment (AJAmran) [reply server/src/controllers/noticeController.ts]: done

**Comment by MD-TANVIR7462** [comment]
Good

**Review by niaz-abir — COMMENTED** [review]
great work

**Review by MD-TANVIR7462 — APPROVED** [review]

**Comment by MD-TANVIR7462** [comment]
Good

**Review by MD-TANVIR7462 — APPROVED** [review]
Good

File: server/src/controllers/noticeController.ts (hunk 3 above)
Comment (niaz-abir) [reply server/src/controllers/noticeController.ts]: ok

**Review by niaz-abir — COMMENTED** [review]
great work

**Comment by jamanador** [comment]
Clean code !!

**Review by jamanador — APPROVED** [review]
Clean Code & Amran was doing great

**Review by UpalBarua — APPROVED** [review]

**Pull request merged**